src/easymotion/easymotion.ts

```diff
@@ -0,0 +1,376 @@
+        totalSteps = Math.floor(totalRemainder / keyTable.length);
+
+        for (var i = 0; i < totalSteps; i++) {
+            keyDepthTable.push(availableKeyTable.pop());
+        }
+    }
+
+    var prefix = "";
+    if (index >= availableKeyTable.length) {
+        // Length of available keys before reset and ";"
+        var oldLength = availableKeyTable.length;
+        // The index that remains after taking away the first-level depth markers
+        var remainder = index - availableKeyTable.length;
+
+        // ";" can be used as the last marker key, when inside a marker with depth. Reset to available keys and add ";"
+        availableKeyTable = keyTable.slice();
+        availableKeyTable.push(";");
+
+        // Depth index counts down instead of up
+        var inverted = (length - oldLength - 1 - remainder);
+        var steps = Math.floor((inverted) / availableKeyTable.length);
+
+        // Add the key to the prefix
+        prefix += keyDepthTable[steps];
+
+        // Check if we're on the last depth level
+        if (steps >= totalSteps) {
+            // Return the proper key for this index
+            return new EasyMotion.Marker(prefix + availableKeyTable[remainder % availableKeyTable.length], markerPosition);
+        }
+
+        // Return the proper index for depths earlier than the last one, including prefix
+        var num = (availableKeyTable.length - 1 - inverted % availableKeyTable.length) % availableKeyTable.length;
+        return new EasyMotion.Marker(prefix + availableKeyTable[num], markerPosition);
+    }
+
+    // Return the last key in the marker, including prefix
+    return new EasyMotion.Marker(prefix + availableKeyTable[index % availableKeyTable.length], markerPosition);
+  }
+
+  /**
+   * Create and cache decoration types for different marker lengths
+   */
+  public static getDecorationType(length: number): vscode.TextEditorDecorationType {
+    var cache = this.decorationTypeCache[length];
+    if (cache) {
+      return cache;
+    }
+
+    var width = length * 8;
+    var type = vscode.window.createTextEditorDecorationType({
+      after: {
+        margin: `0 0 0 -${width}px`,
+        height: `14px`,
+        width: `${width}px`
+      }
+    });
+
+    this.decorationTypeCache[length] = type;
+
+    return type;
+  }
+
+  /**
+   * Create and cache the SVG data URI for different marker codes and colors
+   */
+  private static getSvgDataUri(code: string, backgroundColor: string, fontColor: string): vscode.Uri {
+      var cache = this.svgCache[code];
+      if (cache) {
+        return cache;
+      }
+
+      const width = code.length * 8 + 1;
+      var uri = vscode.Uri.parse(
+        `data:image/svg+xml;utf8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 ${width} ` +
+        `13" height="14" width="${width}"><rect width="${width}" height="14" rx="2" ry="2" ` +
+        `style="fill: ${backgroundColor};"></rect><text font-family="Consolas" font-size="14px" ` +
```

Comment:
@dsifford, Good point, this was the first time I had ever touched anything regarding extensions for VSCode, so I didn't know I could look it up. If someone who has more experience regarding this can fix it that would be great!